src/Grpc.Net.Client/Balancer/BalancerAttributes.cs

```diff
@@ -38,20 +38,22 @@ public sealed class BalancerAttributes : IDictionary<string, object?>, IReadOnly
     /// <summary>
     /// Gets a read-only collection of metadata attributes.
     /// </summary>
-    public static readonly BalancerAttributes Empty = new BalancerAttributes(new ReadOnlyDictionary<string, object?>(new Dictionary<string, object?>()));
```

Comment:
you cite performance as the reason for this change; I would be genuinely surprized if `Dictionary<string,object?` out-performed `ReadOnlyDictionary<string,object?>` except on the custom struct iterator - but if perf is key, I also wonder whether we should be looking to `FrozenDictionary<string, object?>` on TFMs where it is available, via `#if`; is there data behind this change? If the enumerator is the big difference, `FrozenDictrionary<string,object?>` has a custom iterator too, and `foreach` isn't a common scenario on dictionary, usually

Reply:
The performance gain isn't `Dictionary` vs `ReadOnlyDictionary`. The performance gain is from consistently referencing the values as Dictionary instead of IDictionary.

Using `Dictionary` avoids interface overhead and allows `foreach` over the dictionary without allocating.